src/backend/cdb/dispatcher/cdbdisp_async.c

```diff
@@ -822,6 +825,60 @@ handlePollError(CdbDispatchCmdAsync *pParms)
 	return;
 }
 
+/*
+ * Try to completely discard results from a dispatchResult's connection without
+ * extra memory allocations by abusing libpq state-machine hacks.
+ */
+static void
+tryToWipeResults(CdbDispatchResult *dispatchResult)
+{
+	PGresult   *res;
+	PGnotify   *notify;
+	PGconn	   *conn = dispatchResult->segdbDesc->conn;
+
+	/* Free some memory and replace current result with a fatal error dummy. */
+	pqSaveErrorResult(conn);
+
+	/* Make sure PQgetResult() calls are not blocking. */
```

Comment:
Actually, according to the documentation, to check if the `PQgetResult` function is blocked, you need to call the `PQisBusy` function.